### PR TITLE
Implement phone-based training

### PR DESCRIPTION
**Pull request by daemon**
- Add ability to attach phone transcriptions and their alignment from MFA
- Update training and inference routines to support phone-based spotting

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,11 +22,11 @@ In the example that follows, we describe how to train a custom detector for the
 and download an [English pronunciation dictionary](http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b).
 3. Create a positive dataset containing the keyword: 
 ```bash
-INFERENCE_SEQUENCE=[0] DATASET_PATH=data/fire-positive python -m howl.run.create_raw_dataset --negative-pct 0 --vocab fire -i ~/path/to/common-voice --positive-pct 100
+VOCAB='["fire"]' INFERENCE_SEQUENCE=[0] DATASET_PATH=data/fire-positive python -m howl.run.create_raw_dataset --negative-pct 0 -i ~/path/to/common-voice --positive-pct 100
 ```
 4. Create a negative dataset without the keyword:
 ```bash
-INFERENCE_SEQUENCE=[0] DATASET_PATH=data/fire-negative python -m howl.run.create_raw_dataset --negative-pct 5 --vocab fire -i ~/path/to/common-voice --positive-pct 0
+VOCAB='["fire"]' INFERENCE_SEQUENCE=[0] DATASET_PATH=data/fire-negative python -m howl.run.create_raw_dataset --negative-pct 5 -i ~/path/to/common-voice --positive-pct 0
```
Comment (ljj7975) [inline README.md:29]: neg 5% and positive 0%?

File: howl/data/dataset/phone.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+
+__all__ = ['PhoneEnum',
+           'PronunciationDictionary',
+           'PhonePhrase']
+
+
+class PhoneEnum(enum.Enum):
+    SILENCE = 'sil'
+    SILENCE_OPTIONAL = 'sp'
```
Comment (ljj7975) [inline howl/data/dataset/phone.py:15]: what is this for exactly?

File: howl/model/rnn.py [hunk 3]
```diff
@@ -15,6 +15,7 @@
 
 
 class LASEncoderConfig(BaseSettings):
+    num_mels: int = 40
```
Comment (ljj7975) [inline howl/model/rnn.py:18]: back to 40?

File: README.md (hunk 1 above)
Comment (daemon) [reply README.md:29]: 5% of CV is used as the negative set.

File: howl/data/dataset/phone.py (hunk 2 above)
Comment (daemon) [reply howl/data/dataset/phone.py:15]: Those are strings representing the silence and unknown speech phonemes from ARPABET and MFA.

File: howl/model/rnn.py (hunk 3 above)
Comment (daemon) [reply howl/model/rnn.py:18]: It's been at 40.

**Pull request merged**